pyfive/h5d.py

```diff
@@ -84,11 +84,18 @@ def __init__(self, dataobject, pseudo_chunking_size_MB=4):
         self._msg_offset, self.layout_class,self.property_offset = dataobject.get_id_storage_params()
         self._unique = (self._filename, self.shape, self._msg_offset)
 
+        #FIXME: Everyone of these parses the message each time. Optimisation possible?
         if isinstance(dataobject.dtype, tuple):
             if dataobject.dtype[0] == 'ENUMERATION':
                 self._dtype = np.dtype(dataobject.dtype[1], metadata={'enum':dataobject.dtype[2]})
             elif dataobject.dtype[0] == 'COMPOUND':
                 self._dtype = np.dtype(dataobject.dtype[1])
+            elif dataobject.dtype[0] == 'OPAQUE':
+                if dataobject.dtype[1].startswith('NUMPY:'):
+                    self._dtype = np.dtype(dataobject.dtype[1][6:], metadata={'h5py_opaque': True})
+                else: 
+                    # Can't test this, since I don't know how to write a test dataset
+                    self._dtype = np.dtype('V'+str(dataobject.dtype[1]), metadata={'h5py_opaque': True})
```

Comment:
This is connected to my other question about possible missing `size`. `dataobject.dtype[1]` should contain the size here?

Reply:
This should write an opaque dataset to test with:

```python
import h5py
import numpy as np

with h5py.File("opaque_fixed.h5", "w") as f:
    # Define a fixed-size opaque dtype as NumPy void
    max_len = 64  # bytes per element
    dt = np.dtype(f"V{max_len}")  # 'V' = void type

    # Create dataset
    dset = f.create_dataset("opaque_data", shape=(3,), dtype=dt)

    data = [
        b"hello world",
        b"\x01\x02\x03\x04custombinarydata",
        bytes(range(10))
    ]

    for i, blob in enumerate(data):
        buf = blob[:max_len].ljust(max_len, b'\x00')
        dset[i] = np.void(buf)
```

Reply:
Hmm. Ok. Maybe there is a role for size. Will put this in and see. Thanks.